internal/freelist/shared.go

```diff
@@ -67,9 +67,6 @@ func (t *shared) Free(txid common.Txid, p *common.Page) {
 	allocTxid, ok := t.allocs[p.Id()]
 	if ok {
 		delete(t.allocs, p.Id())
-	} else if p.IsFreelistPage() {
-		// Freelist is always allocated by prior tx.
-		allocTxid = txid - 1
```

Comment:
The Line 79 `txp.alloctx = append(txp.alloctx, allocTxid)` could append zero into alloctx.